pageserver/src/page_service.rs

```diff
@@ -1326,24 +1330,29 @@ where
                 .for_command(ComputeCommandKind::Basebackup)
                 .inc();
 
-            let (lsn, gzip) = match (params.get(2), params.get(3)) {
-                (None, _) => (None, false),
-                (Some(&"--gzip"), _) => (None, true),
-                (Some(lsn_str), gzip_str_opt) => {
-                    let lsn = Lsn::from_str(lsn_str)
-                        .with_context(|| format!("Failed to parse Lsn from {lsn_str}"))?;
-                    let gzip = match gzip_str_opt {
-                        Some(&"--gzip") => true,
-                        None => false,
-                        Some(third_param) => {
-                            return Err(QueryError::Other(anyhow::anyhow!(
-                                "Parameter in position 3 unknown {third_param}",
-                            )))
+            let mut lsn = None;
+            let mut replica = false;
+            let mut gzip = false;
+            for i in 2..params.len() {
```

Comment:
The for with the if could be written more compactly with a `for param in &params[2..]`.

Reply:
Thank you for suggestion!
Done